pkg/cloud/volume_limits.go

```diff
@@ -131,215 +131,337 @@ func GetNVMeInstanceStoreVolumesForInstanceType(it string) int {
 	return 0
 }
 
-// / https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/InstanceStorage.html#instance-store-volumes
+// / https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-store-volumes.html
 // / IMDS does not provide NVMe instance store data; we'll just list all instances here
 // / TODO: See if we can get these values from DescribeInstanceTypes API
+// Generated by command, requires aws cli and jq:
+// aws ec2 describe-instance-types --region us-east-1 --filters "Name=instance-storage-supported,Values=true" --query "InstanceTypes[].[InstanceType, InstanceStorageInfo]" | jq -r 'map("\"" + .[0] + "\": " + (.[1].Disks | map(.Count) | add | tostring) + ",") | .[]' | sort
```

Comment:
Nit: Shall we consider using `awk` to make sure the sort mostly preserves the original, cleaner to read order? 

`awk -F'[:,]' '{ print $1, $2, $3 }' | awk -F'.' '{ print $1, $2, $3 }' | sort -k1,1 -k2,2n | awk '{ printf "%s.%s: %s\n", $1, $2, $3 }`

`aws ec2 describe-instance-types --region us-east-1 --filters "Name=instance-storage-supported,Values=true" --query "InstanceTypes[].[InstanceType, InstanceStorageInfo]" | jq -r 'map("\"" + .[0] + "\": " + (.[1].Disks | map(.Count) | add | tostring) + ",") | .[]' | awk -F'[:,]' '{ print $1, $2, $3 }' | awk -F'.' '{ print $1, $2, $3 }' | sort -k1,1 -k2,2n | awk '{ printf "%s.%s: %s\n", $1, $2, $3 }'`

The resulting list gets sorted like so, which is a bit nicer to read. 

```
...
"x2gd.large": 1
"x2gd.medium": 1
"x2gd.metal": 2
"x2gd.xlarge": 1
"x2gd.2xlarge": 1
"x2gd.4xlarge": 1
"x2gd.8xlarge": 1
"x2gd.12xlarge": 2
"x2gd.16xlarge": 2
"x2idn.metal": 2
"x2idn.16xlarge": 1
"x2idn.24xlarge": 2
"x2idn.32xlarge": 2
"x2iedn.metal": 2
"x2iedn.xlarge": 1
"x2iedn.2xlarge": 1
"x2iedn.4xlarge": 1
"x2iedn.8xlarge": 1
"x2iedn.16xlarge": 1
"x2iedn.24xlarge": 2
"x2iedn.32xlarge": 2
"z1d.large": 1
"z1d.metal": 2
"z1d.xlarge": 1
"z1d.2xlarge": 1
"z1d.3xlarge": 1
"z1d.6xlarge": 1
"z1d.12xlarge": 2
```

CC @torredil

Reply:
Given that this is auto-generated, I don't think there's any advantage to spending extra complexity making it human readable - humans aren't intended to read it on any regular basis.